### ISSUE TITLE
v-model reactivity issue when iterating through an object

### ISSUE THREAD

**Issue by benceg** [yyx990803/vue#689]
```javascript
this.$add('item', { "foo" : "x", "bar" : "y" });
```

```html
<li v-repeat="item">
  <label>{{ $key }}</label>
  <input v-model="item[$key]">
</li>
```

I've snagged some quirky behavior when iterating through an object and referencing its model in this way. This will get reactivity working for item[$key], and is the only way to do so, but the input will blur on keypress unless the `lazy` attribute is supplied.

This is also counterpoint to the array use case, where a key can be supplied for v-model. Using `v-model="$key"` in this instance just renders the `$key`'s name into the input.

I've also noticed that, when `item` is filled with empty values, blurring a field will copy the entire field value across to the next model (if using `lazy`) or skip the cursor to the next input field.

That aside, loving Vue, thanks so much for it!

**Comment by TMiguelT** [comment]
My suspicion is that your inputs are blurring because they're actually being deleted and then recreated in the next frame, which means they lose focus. This is just a guess though, and it would mean that Vue re-runs the entire `v-repeat` just because you change one object element, which seems unnecessary. 

Anyway here's a fiddle for anyone who wants to look into this further: http://jsfiddle.net/unr38q1b/1/

**Comment by benceg** [comment]
Thank you for the CodePen @TMiguelT! I think your suspicion is absolutely correct.

**Comment by benceg** [comment]
Could anyone perhaps lend a hand with this? To my discredit I don't believe I know enough about ES5 getters & setters and Vue's internal implementation of them to be useful in this situation.

**Comment by yyx990803** [comment]
This is because Vue converts the object into a fresh array on update, so `v-repeat` can't do in place updates because the array identity changed. Maybe diffing the actual object will be a better solution.

A work around is converting the object into an array yourself, then use `v-repeat` on that array instead. If you need data persistence you can convert that array back to an object. This doesn't sound ideal but should work.

**Comment by benceg** [comment]
Thank you @yyx990803.

I did end up implementing your suggested workaround a while ago w/additional parsing for data persistence. Luckily the nature of the data itself is not completely arbitrary so it worked out just fine. I'm happy to close this thread unless you'd prefer to leave it open for future implementation purposes.

**Comment by yyx990803** [comment]
This has been fixed in 0.12 beta!

**Comment by jonathanpmartins** [comment]
Nice feature! Using 0.12 beta like so v-model="array[$index].mycustomkey". All working nice!
